includes/Core/Key_Metrics/Key_Metrics_Settings.php

```diff
@@ -49,6 +49,37 @@ protected function get_default() {
 		return array();
 	}
 
+	/**
+	 * Merges an array of settings to update. Only existing keys will be updated.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $partial Partial settings array to save.
+	 * @return bool True on success, false on failure.
+	 */
+	public function merge( array $partial ) {
+		$settings = $this->get();
+		if ( empty( $settings ) ) {
+			// Set defaults just to make sure that only allowed settings
+			// are merged from the $partial array.
+			$settings = array(
+				'widgetSlugs'    => '',
+				'isWidgetHidden' => '',
+			);
+		}
+
+		$partial = array_filter(
+			$partial,
+			function ( $value ) {
+				return null !== $value;
+			}
+		);
+
+		$updated = array_intersect_key( $partial, $settings );
```

Comment:
Hi @eugene-manuilov, thanks for making changes here, however IMHO it's still not quite right.

Due to setting these default values, if `merge` is called without `isWidgetHidden`, the result of calling `get_sanitize_callback` on the empty string value will end up being set, in other words `isWidgetHidden` will default to `false`.

It's also a bit unclear to set default values which are not the expected data type of the settings. If we _do_ want to default `isWidgetHidden` then I'd make it explicit here by setting it to `false` instead of the empty string. However, I'd suggest that instead, we don't set defaults and take an approach like the following:

```suggestion
		$partial = array_filter(
			$partial,
			function ( $value ) {
				return null !== $value;
			}
		);

		$allowed_settings = array(
			'widgetSlugs'    => true,
			'isWidgetHidden' => true,
		);

		$updated = array_intersect_key( $partial, $allowed_settings );
```